src/logic/shorten-url.js

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import fetch from 'node-fetch';
+
+import { config } from '../config';
+import { getLogger } from '../log';
+
+class BitlyConfigurationError extends Error {
+  name = 'BitlyConfigurationError';
+  status = 500;
+  expose = true; // The message will be exposed to users
+}
+
+export async function shortenUrl(longUrl: string): Promise<string> {
+  const log = getLogger('shortenUrl');
+  if (!config.bitlyToken) {
+    log.error('No access token for bitly has been configured!');
+    throw new BitlyConfigurationError(
+      'No access token for bitly has been configured!'
+    );
+  }
+
+  const bitlyQueryUrl = 'https://api-ssl.bitly.com/v4/shorten';
+  const payload = { long_url: longUrl };
+
+  const response = await fetch(bitlyQueryUrl, {
+    method: 'POST',
+    headers: {
+      Authorization: `Bearer ${config.bitlyToken}`,
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(payload),
+  });
+
+  if (!response.ok) {
+    throw new Error(
```

Comment:
Also, it might be good to try and read the error message, and log it or report it. I could see an error class do something like `new BitlyErrorResponse(response)`, then it could attempt to read the error message a bit more in-depth:

<img width="349" alt="image" src="https://user-images.githubusercontent.com/1588648/77440751-a8b2ff80-6db6-11ea-92e0-2c8178fa9b70.png">

https://dev.bitly.com/v4/#operation/createBitlink

Reply:
yeah, good idea, I'll dig deeper.